modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -1625,6 +1625,71 @@ public void deleteFlowchart(String name, String version, String user) throws Pro
         }
     }
 
+    /**
+     * Delete the resource collection (directory) created in the governance registry for the process to keep its documents
+     *
+     * @param processName
+     * @param processVersion
+     * @throws ProcessCenterException
+     */
+    public void deleteDocumentResourceCollection(String processName, String processVersion)
+            throws ProcessCenterException, RegistryException {
+
+        RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+        String documentsResourceCollectionPath =
+                ProcessCenterConstants.DOC_CONTENT_PATH + processName + "/" + processVersion;
+        if (registryService != null) {
+            UserRegistry reg = registryService.getGovernanceSystemRegistry();
+            reg.delete(documentsResourceCollectionPath);
+        }
+    }
+
+    /**
+     * Delete process related text
+     *
+     * @param processName
+     * @param processVersion
+     * @return
+     * @throws ProcessCenterException
+     */
+    public String deleteProcessText(String processName, String processVersion) throws ProcessCenterException {
+        String processId = "";
+        try {
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+            if (registryService != null) {
+                UserRegistry reg = registryService.getGovernanceSystemRegistry();
+                String processTextPath = ProcessCenterConstants.PROCESS_TEXT_PATH + processName + "/" + processVersion;
+                if (reg.resourceExists(processTextPath)) {
+                    //delete the processText resource
+                    reg.delete(processTextPath);
+                }
+
+                //delete the processText resource's path defined in process.rxt
+                String processPath = ProcessCenterConstants.PROCESS_ASSET_ROOT + processName + "/" + processVersion;
+                if (reg.resourceExists(processPath)) {
+                    Resource processResource = reg.get(processPath);
+
+                    String processContent = new String((byte[]) processResource.getContent());
+                    Document processXML = stringToXML(processContent);
+                    processXML.getElementsByTagName("processtextpath").item(0).setTextContent("NA");
+
+                    String newProcessContent = xmlToString(processXML);
+                    processResource.setContent(newProcessContent);
+                    reg.put(processPath, processResource);
+
+                    Resource storedProcessAsset = reg.get(processPath);
+                    processId = storedProcessAsset.getUUID();
+                }
+            }
+        } catch (Exception e) {
+            String errMsg = "Error has been occurred while removing BPMN diagram in the process:" + processName + "-"
```

Comment:
Thanks @mohanvive. The xmlToString method called in the try block throws 'Exception'. So has to catch 'Exception'. And the issue related to logging and throwing exception same time, is fixed already with 636d316f1c9c92034a9d3c7c1a8fef26dbd5ba1b